api/v1alpha1/author_webhook.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"github.com/redhat-appstudio/release-service/metadata"
+	admissionv1 "k8s.io/api/admission/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const MAX_LABEL_LENGTH = 63
+
+// authorWebhook describes the data structure for the author webhook
+type authorWebhook struct {
+	Client client.Client
+	log    *logr.Logger
+}
+
+// +kubebuilder:webhook:path=/mutate-appstudio-redhat-com-v1alpha1-author,mutating=true,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releases;releaseplans,verbs=create;update,versions=v1alpha1,name=mauthor.kb.io,admissionReviewVersions=v1
+
+// RegisterAuthorWebhook registers the author webhook with the passed manager and log.
+func RegisterAuthorWebhook(mgr ctrl.Manager, log logr.Logger) {
+	mgr.GetWebhookServer().Register("/mutate-appstudio-redhat-com-v1alpha1-author", &webhook.Admission{
+		Handler: &authorWebhook{
+			Client: mgr.GetClient(),
+			log:    &log,
+		},
+	})
+}
+
+// Handle creates an admission response for Release and ReleasePlan requests.
+func (a *authorWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
Maybe this function could do something like this:
```
func (a *authorWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
        var object client.Object
	err := a.decoder.Decode(req, object)
	if err != nil {
		return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, "error decoding request"))
	}

	switch req.Kind.Kind {
	case "Release":
		return a.handleRelease(req, object.(*v1alpha1.Release))
	case "ReleasePlan":
		return a.handleReleasePlan(req, object.(*v1alpha1.ReleasePlan))
	}
}
```
That way you already have the `req` with the `req.Object.Raw` and the object itself.